lib/storage/src/content_manager/toc/collection_meta_ops.rs

```diff
@@ -355,27 +356,30 @@ impl TableOfContent {
                     )));
                 };
 
-                // Transfer must have changed configuration
-                if transfers.contains(&transfer) {
+                if old_transfer.method == Some(transfer_restart.method) {
                     return Err(StorageError::bad_request(format!(
                         "Cannot restart transfer for shard {} from {} to {}, its configuration did not change",
-                        transfer.shard_id, transfer.from, transfer.to,
+                        transfer_restart.shard_id, transfer_restart.from, transfer_restart.to,
                     )));
                 }
 
                 // Abort and start transfer
                 self.handle_transfer(
                     collection_id.clone(),
                     ShardTransferOperations::Abort {
-                        transfer: transfer.key(),
+                        transfer: transfer_restart.key(),
                         reason: "restart transfer".into(),
                     },
                 )
                 .await?;
 
-                // Preserve sync flag from the old or new transfer
-                let mut new_transfer = transfer.clone();
-                new_transfer.sync = old_transfer.sync || new_transfer.sync;
+                let new_transfer = ShardTransfer {
+                    shard_id: transfer_restart.shard_id,
+                    from: transfer_restart.from,
+                    to: transfer_restart.to,
+                    sync: old_transfer.sync, // Preserve sync flag from the old transfer
```

Comment:
Just to clarify, this is desired, intended behavior, right?

(I didn't dig into it deep enough, I have a very vague idea what's going on. Just clarifying, cause it's been changed there and back a few times.)

@generall @timvisee

Reply:
Ok, I think I get it now.

Before we reused `ShardTransfer` struct that had `sync` flag, and now we use `RestartShardTransfer`, so there's no second `sync` flag, so we just use `old_transfer.sync` here.